numerics/elliptic_integrals.cpp

```diff
@@ -1744,27 +1802,63 @@ void FukushimaEllipticBDJ(Angle const& φ,
     // See [Fuk11b], equations (B.2), and [Fuk12], equation (A.2).
     B_φǀm += 2 * j * B;
     D_φǀm += 2 * j * D;
-    J_φ_nǀm += 2 * j * J;
+    if constexpr (should_compute<ThirdKind>) {
+      J_φ_nǀm += 2 * j * J;
+    }
+  }
+}
+
+template<typename ThirdKind, typename>
+void EllipticFEΠ(Angle const& φ,
+                 double const n,
+                 double const mc,
+                 Angle& F_φǀm,
+                 Angle& E_φǀm,
+                 ThirdKind& Π_φ_nǀm) {
+  DCHECK_LE(0, n);
+  DCHECK_GE(1, n);
+  DCHECK_LE(0, mc);
+  DCHECK_GE(1, mc);
+  Angle B{uninitialized};
+  Angle D{uninitialized};
+  ThirdKind J{uninitialized};
+  FukushimaEllipticBDJ(φ, n, mc, B, D, J);
+  F_φǀm = B + D;
+  E_φǀm = B + mc * D;
+  if constexpr (should_compute<ThirdKind>) {
+    Π_φ_nǀm = F_φǀm + n * J;
   }
 }
 
+}  // namespace
+
+void FukushimaEllipticBDJ(Angle const& φ,
+                          double const n,
+                          double const mc,
+                          Angle& B_φǀm,
+                          Angle& D_φǀm,
+                          Angle& J_φ_nǀm) {
+  return FukushimaEllipticBDJ<Angle>(φ, n, mc, B_φǀm, D_φǀm, J_φ_nǀm);
```

Comment:
Is the specification of `Angle` necessary?  I would have expected that it would deduce it.

Reply:
It would, but the non-templatized `FukushimaEllipticBDJ` (which we are defining) has precedence here.

Reply:
An alternative would be to expose `UnusedResult` in `base`, and to only have the templatized version (declared in the header, with both explicit instantiation declarations).